Source/Core/Core/AchievementManager.cpp

```diff
@@ -1559,14 +1573,36 @@ void AchievementManager::HandleLeaderboardStartedEvent(const rc_runtime_event_t*
     {
       OSD::AddMessage(fmt::format("Attempting leaderboard: {}", m_game_data.leaderboards[ix].title),
                       OSD::Duration::VERY_LONG, OSD::Color::GREEN);
+      LeaderboardDisplay display{.id = runtime_event->id,
+                                 .format = m_game_data.leaderboards[ix].format};
+      rc_runtime_format_lboard_value(display.value.data(), FORMAT_SIZE, runtime_event->value,
+                                     display.format);
+      m_active_leaderboards.push_back(display);
       return;
     }
   }
   ERROR_LOG_FMT(ACHIEVEMENTS, "Invalid leaderboard started event with id {}.", runtime_event->id);
 }
 
+void AchievementManager::HandleLeaderboardUpdatedEvent(const rc_runtime_event_t* runtime_event)
+{
+  for (LeaderboardDisplay& display : m_active_leaderboards)
+  {
+    if (display.id == runtime_event->id)
+    {
+      rc_runtime_format_lboard_value(display.value.data(), FORMAT_SIZE, runtime_event->value,
+                                     display.format);
+      return;
+    }
+  }
+  ERROR_LOG_FMT(ACHIEVEMENTS, "Invalid leaderboard updated event with id {}.", runtime_event->id);
+}
+
 void AchievementManager::HandleLeaderboardCanceledEvent(const rc_runtime_event_t* runtime_event)
 {
+  for (auto itx = m_active_leaderboards.begin(); itx < m_active_leaderboards.end(); itx++)
```

Comment:
Ditto wrt the `erase_if` suggestion (forgot to comment here in the original review)